src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const originalDate = date.split(fromSeparator);
+
+  const newDate = [];
+  const dateObj = {
+
+  };
+
+  for (let i = 0; i < 3; i++) {
+    dateObj[fromFormat[i]] = originalDate[i];
+
+    if (fromFormat[i] === 'YY') {
+      if (+originalDate[i] < 30) {
+        dateObj.YYYY = 20 + originalDate[i];
+      } else {
+        dateObj.YYYY = 19 + originalDate[i];
+      }
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
+    newDate[i] = dateObj[toFormat[i]];
+
+    if (toFormat[i] === 'YY') {
+      newDate[i] = dateObj.YYYY.slice(2);
+    } else if (toFormat[i] === 'YYYY') {
+      newDate[i] = dateObj.YYYY;
+    }
```

Comment:
You can use `toFormat[i].length` and simplify expresion to avoid code duplicate
```suggestion
    if (toFormat[i] === 'YY') {
      newDate[i] = dateObj.YYYY.slice(2);
    } else if (toFormat[i] === 'YYYY') {
      newDate[i] = dateObj.YYYY;
    }
```